File: app/src/main/java/com/nacare/capture/data/service/network/BasicAuthInterceptor.java
package com.nacare.capture.data.service.network;

import java.io.IOException;
import java.util.Base64;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class BasicAuthInterceptor implements Interceptor {

    private final String username;
    private final String password;

    public BasicAuthInterceptor(String username, String password) {
        this.username = username;
        this.password = password;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        String credentials = username + ":" + password;
        String basicAuth = "Basic " + Base64.getEncoder().encodeToString(credentials.getBytes());

        Request request = chain.request().newBuilder()
                .header("Authorization", basicAuth)
                .build();

        return chain.proceed(request);
    }
}

File: app/src/main/java/com/capture/app/network/BasicAuthInterceptor.kt
package com.capture.app.network
import okhttp3.Interceptor
import okhttp3.Response


class BasicAuthInterceptor(private val encodedString: String) :
    Interceptor {

    override fun intercept(chain: Interceptor.Chain): Response {

        val basicAuth = "Basic $encodedString"

        val request = chain.request().newBuilder()
            .header("Authorization", basicAuth)
            .build()

        return chain.proceed(request)
    }
}